packages/xstate-react/package.json

```diff
@@ -54,7 +54,7 @@
     "url": "https://github.com/statelyai/xstate/issues"
   },
   "peerDependencies": {
-    "react": "^16.8.0 || ^17.0.0 || ^18.0.0",
+    "react": "^16.8.0 || ^17.0.0 || ^18.0.0 || ^19.0.0",
```

Comment:
I wonder - what motivated this change now? If I'm not mistaken there is no version out there that's compatible with what is added here, so in some sense - this is going to be a "noop" until React 19 actually gets released.

Reply:
I’m not sure about the details of package fixes, but here’s what I have encountered as a XState user.

The recently released Next.js 15 supports React 19. However, when installing `@xstate/react`, I have encountered peer dependency errors. Judging from the logs, it seems `@xstate/react` does not support React 19. To suppress the error message, I needed to downgrade `react` and `react-dom` from 19 to 18.

https://nextjs.org/blog/next-15

```bash
npx create-next-app@latest next-xstate-15 \
                              --typescript \
                              --eslint \
                              --import-alias "@/*" \
                              --src-dir \
                              --use-pnpm \
                              --tailwind \
                              --app \
                              --use-pnpm \
                              --turbopack
cd next-xstate-15
```

Install `xstate`:


```bash
pnpm install xstate 
```
```bash
dependencies:
+ xstate 5.18.2

Done in 748ms
```

When installing `@xstate/react`, the error message indicates that `@xstate/react` does not support React 19:

```bash
pnpm install @xstate/react
```

```bash
dependencies:
+ @xstate/react 4.1.3

 WARN  Issues with peer dependencies found
.
└─┬ @xstate/react 4.1.3
  ├── ✕ unmet peer react@"^16.8.0 || ^17.0.0 || ^18.0.0": found 19.0.0-rc-69d4b800-20241021
  ├─┬ use-isomorphic-layout-effect 1.1.2
  │ └── ✕ unmet peer react@"^16.8.0 || ^17.0.0 || ^18.0.0": found 19.0.0-rc-69d4b800-20241021
  └─┬ use-sync-external-store 1.2.2
    └── ✕ unmet peer react@"^16.8.0 || ^17.0.0 || ^18.0.0": found 19.0.0-rc-69d4b800-20241021

Done in 744ms
```

Here is `package.json`:

```json:package.json
{
  ...
  "dependencies": {
    "@xstate/react": "^4.1.3",
    "next": "15.0.1",
    "react": "19.0.0-rc-69d4b800-20241021",
    "react-dom": "19.0.0-rc-69d4b800-20241021",
    "xstate": "^5.18.2"
  },
  ...
}
```

Here is my repository:

https://github.com/hayato94087/next-xstate-15

Reply:
@Andarist See here for the original discussion: https://discord.com/channels/795785288994652170/1299849094629560421/1299849094629560421

Reply:
The problem is that this doesn't solve this issue:
```ts
require('semver').satisfies('19.0.0-rc-69d4b800-20241021', '^19.0.0') // false
```

To make it work you have to relax this peer dep range even further to allow prereleases